core/common/src/main/java/alluxio/network/protocol/databuffer/NioHeapBufferPool.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.network.protocol.databuffer;
+
+import alluxio.exception.runtime.ResourceExhaustedRuntimeException;
+import alluxio.retry.RetryPolicy;
+
+import java.nio.ByteBuffer;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Heap buffer pool.
+ */
+public class NioHeapBufferPool {
+  private static final TreeMap<Integer, LinkedList<ByteBuffer>> BUF_POOL = new TreeMap<>();
```

Comment:
The current memory pool is only used by CRC, and the application will not be recycled. Will it occupy core business resources? I think CRC verification is a low-frequency non-core business that does not require high efficiency. Should I directly apply for heap memory temporarily? That’s it

Reply:
i added a non-pooled option